src/front/wgsl/mod.rs

```diff
@@ -1092,32 +1093,31 @@ impl Parser {
                     }
                     lexer.expect(Token::Paren(')'))?;
                     let expr = if components.is_empty() {
-                        let last_component_inner = ctx.resolve_type(last_component)?;
-                        match (&inner, last_component_inner) {
-                            (
-                                &crate::TypeInner::Scalar { .. },
-                                &crate::TypeInner::Scalar { .. },
-                            )
-                            | (
-                                &crate::TypeInner::Matrix { .. },
-                                &crate::TypeInner::Matrix { .. },
-                            )
-                            | (
-                                &crate::TypeInner::Vector { .. },
-                                &crate::TypeInner::Vector { .. },
-                            ) => crate::Expression::As {
-                                expr: last_component,
-                                kind: kind.ok_or(Error::BadTypeCast(word))?,
-                                convert: true,
-                            },
+                        ctx.resolve_type(last_component)?;
```

Comment:
why are we not using the result of this?

Reply:
I left a comment in the rebased code. Short version is that `resolve_type` mutably borrows ctx, but we need an immutable borrow of it, if we're going to borrow it to look up `handle` as well.